lib/kubernetes-deploy/ejson_secret_provisioner.rb

```diff
@@ -0,0 +1,153 @@
+# frozen_string_literal: true
+require 'json'
+require 'base64'
+require 'open3'
+require 'kubernetes-deploy/kubeclient_builder'
+require 'kubernetes-deploy/logger'
+
+module KubernetesDeploy
+  class EjsonSecretError < FatalDeploymentError
+    def initialize(msg)
+      super("Creation of Kubernetes secrets from ejson failed: #{msg}")
+    end
+  end
+
+  class EjsonSecretProvisioner
+    include KubernetesDeploy::KubeclientBuilder
+
+    MANAGEMENT_ANNOTATION = "kubernetes-deploy.shopify.io/ejson-secret".freeze
+    MANAGED_SECRET_EJSON_KEY = "kubernetes_secrets".freeze
+    EJSON_SECRETS_FILE = "secrets.kubernetes-deploy.ejson".freeze
+    EJSON_KEYS_SECRET = "ejson-keys".freeze
+
+    def initialize(namespace:, context:, template_dir:, client: nil)
+      @namespace = namespace
+      @context = context
+      @ejson_file = "#{template_dir}/#{EJSON_SECRETS_FILE}"
+      @kubeclient = client || build_v1_kubeclient(@context)
+    end
+
+    def secrets_requested?
```

Comment:
From the content of the method I think `secrets_provised?` would better describe it. What do you think?